opendatahub-io/data-science-pipelines-operator
# Expose API Server and Envoy endpoints and add a new Envoy condition

## The issue resolved by this Pull Request:
Resolves [RHOAIENG-7929](https://issues.redhat.com/browse/RHOAIENG-7929)

## Description of your changes:
This PR will introduce a new condition to check if MLMD Envoy is ready (EnvoyReady) and add the DSPA endpoints (API Server and MLMD Envoy) to the status field.

## Testing instructions
* Build the operator pod, or use the one that GitHub will create
* Deploy DSPO:
 ```
export IMAGE_NAME="" # Change to the image you built or the one GitHub generated
make deploy IMG=$IMAGE_NAME OPERATOR_NS=opendatahub # Ensure you have opendatahub namespace created and no resources created in it
```
* Deploy the [dspa_simple.yaml](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/samples/v2/dspa-simple/dspa_simple.yaml) file
* Run the following command (ensure you ran `oc login` first):
```
oc get DataSciencePipelinesApplication sample -o yaml | yq .status.components
```
* You should receive an output like this:
```
apiServer:
  externalUrl: https://ds-pipeline-sample-dspa.apps.example.com
  url: https://ds-pipeline-sample.dspa.svc.cluster.local:8443
mlmdProxy:
  externalUrl: https://ds-pipeline-md-sample-dspa.apps.example.com
  url: http://ds-pipeline-md-sample.dspa.svc.cluster.local:9090
```
* Ensure unit tests are passing. Run the following in the repo root path:
```
make unittests
```

## Checklist
